pkg/frontend/metrics.go

```diff
@@ -0,0 +1,35 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"time"
+
+	"github.com/Azure/ARO-RP/pkg/util/recover"
+)
+
+//emitMetrics sends a heartbeat metric every 60 seconds
+func (f *frontend) emitMetrics(stop <-chan struct{}) {
+	defer recover.Panic(f.baseLog)
+	t := time.NewTicker(time.Minute)
+	defer t.Stop()
+	f.baseLog.Print("starting heartbeat")
+
+	for range t.C {
+		select {
+		case <-stop:
+			return
+		default:
```

Comment:
use the same pattern that's everywhere else in the code, take care with reinvention.  As you have this implemented currently, you could be blocked for up to a minute before noticing that stop has been closed.  Do this:

```
for {
	// do stuff

	select {
	case <-t.C:
	case <-stop:
		return
	}
}
```